File: Backend/retail/src/main/java/com/example/retail/service/CartService.java
package com.example.retail.service;

@Service
@RequiredArgsConstructor
public class CartService {

    private final CartRepository cartRepo;
    private final MenuRepository menuRepo;
    private final UserRepository userRepo;

    public CartItem addToCart(Long userId, Long menuId, int qty) {

        User user = userRepo.findById(userId).orElseThrow();
        MenuItem item = menuRepo.findById(menuId).orElseThrow();

        CartItem cart = new CartItem();
        cart.setUser(user);
        cart.setMenuItem(item);
        cart.setQuantity(qty);

        return cartRepo.save(cart);
    }

    public List<CartItem> getCart(Long userId) {
        User user = userRepo.findById(userId).orElseThrow();
        return cartRepo.findByUser(user);
    }

    public CartItem update(Long id, int qty) {
        CartItem cart = cartRepo.findById(id).orElseThrow();
        cart.setQuantity(qty);
        return cartRepo.save(cart);
    }

    public void delete(Long id) {
        cartRepo.deleteById(id);
    }
}

File: Frontend/retail-app/src/services/CartService.js
import axios from "axios";

const API = "http://localhost:8080/api/cart";

export const addToCart = (data) => axios.post(API, data);

export const getCart = (userId) =>
    axios.get(`${API}?userId=${userId}`);

export const updateCart = (id, qty) =>
    axios.put(`${API}/${id}?qty=${qty}`);

export const deleteCart = (id) =>
    axios.delete(`${API}/${id}`);